listenbrainz/troi/import_ms.py

```diff
@@ -0,0 +1,73 @@
+import spotipy
+import requests
+from troi import Recording
+import troi.playlist
+from troi.musicbrainz.recording import RecordingListElement
+import sys
+
+def import_from_spotify(token, user, playlist_id):
```

Comment:
Not sure exactly what the issue is, but I deployed the PR on test.LB and I couldn't import a playlist.
One issue is clear regarding rate-limiting, considering the flurry of 429 error codes
 
Here are the container logs for one import attempt:
```
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
Error occurred: 429
2024-03-19 16:56:48,701 listenbrainz.webserver ERROR    Exception on /1/playlist/spotify/6oSNou76m5UwIpe9PH8vm4/tracks [GET]
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/code/listenbrainz/listenbrainz/webserver/decorators.py", line 18, in decorator
    resp = make_response(f(*args, **kwargs))
                         ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/brainzutils/ratelimit.py", line 262, in rate_limited
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/code/listenbrainz/listenbrainz/webserver/decorators.py", line 42, in decorator
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/code/listenbrainz/listenbrainz/webserver/views/playlist_api.py", line 902, in import_tracks_from_spotify_to_playlist
    playlist = import_from_spotify(token["access_token"], user["auth_token"], playlist_id)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/code/listenbrainz/listenbrainz/troi/import_ms.py", line 26, in import_from_spotify
    recordings = [Recording(mbid=track["recording_mbid"]) for track in mbid_mapped_tracks if "recording_mbid" in track]
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/code/listenbrainz/listenbrainz/troi/import_ms.py", line 26, in <listcomp>
    recordings = [Recording(mbid=track["recording_mbid"]) for track in mbid_mapped_tracks if "recording_mbid" in track]
                                                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: argument of type 'NoneType' is not iterable

```

Reply:
Hmm, its possible that the endpoint is returning None for tracks which couldn't be found in the mapping/cache.